src/test/java/org/mule/service/http/impl/functional/server/HttpBrokenPipeLoggingConfigTestCase.java

```diff
@@ -0,0 +1,172 @@
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.service.http.impl.functional.server;
+
+import static org.mule.runtime.api.metadata.MediaType.TEXT;
+import static org.mule.runtime.core.api.util.ClassUtils.setContextClassLoader;
+import static org.mule.runtime.http.api.HttpConstants.HttpStatus.INTERNAL_SERVER_ERROR;
+import static org.mule.runtime.http.api.HttpConstants.Method.POST;
+import static org.mule.runtime.http.api.HttpHeaders.Names.CONNECTION;
+import static org.mule.runtime.http.api.HttpHeaders.Names.CONTENT_TYPE;
+import static org.mule.runtime.http.api.HttpHeaders.Values.CLOSE;
+
+import static java.lang.String.format;
+import static java.lang.Thread.currentThread;
+import static java.util.Collections.singletonList;
+
+import static com.github.valfirst.slf4jtest.TestLoggerFactory.getTestLogger;
+import static org.apache.http.entity.ContentType.DEFAULT_TEXT;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+import org.mule.runtime.core.internal.util.CompositeClassLoader;
+import org.mule.runtime.http.api.domain.entity.InputStreamHttpEntity;
+import org.mule.runtime.http.api.domain.message.response.HttpResponse;
+import org.mule.runtime.http.api.domain.request.HttpRequestContext;
+import org.mule.runtime.http.api.server.RequestHandler;
+import org.mule.runtime.http.api.server.async.HttpResponseReadyCallback;
+import org.mule.runtime.http.api.server.async.ResponseStatusCallback;
+import org.mule.service.http.impl.service.server.grizzly.BaseResponseCompletionHandler;
+import org.mule.tck.probe.JUnitLambdaProbe;
+import org.mule.tck.probe.PollingProber;
+
+import java.io.ByteArrayInputStream;
+import java.net.SocketTimeoutException;
+import java.util.concurrent.CountDownLatch;
+
+import com.github.valfirst.slf4jtest.TestLogger;
+import io.qameta.allure.Description;
+import org.apache.http.client.fluent.Request;
+import org.junit.Before;
+import org.junit.Test;
+
+public class HttpBrokenPipeLoggingConfigTestCase extends AbstractHttpServerTestCase {
+
+  private CountDownLatch requestLatch;
+  private CountDownLatch responseLatch;
+  private ClassLoader requestHandlerClassLoader;
+
+  TestLogger testLogger = getTestLogger(BaseResponseCompletionHandler.class);
+
+  public HttpBrokenPipeLoggingConfigTestCase(String serviceToLoad) {
+    super(serviceToLoad);
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    requestLatch = new CountDownLatch(1);
+    responseLatch = new CountDownLatch(1);
+
+    setUpServer();
+
+    // Changing the classloader in order to imitate the app's behaviour when handling the error
+    Thread currentThread = currentThread();
+    ClassLoader originalClassLoader = currentThread.getContextClassLoader();
+    requestHandlerClassLoader = CompositeClassLoader.from(currentThread().getContextClassLoader());
```

Comment:
This one could be statically imported. But I'm not sure about how readable it would be when we remove the `CompositeClassLoader` qualifier from the line:
`requestHandlerClassLoader = from(currentThread().getContextClassLoader());`
The information about which `from` we are using here is lost in the nearby context and would require analysis of the imports at the top of the file.
For that reason I think it is best to leave it as it is.